km/km_vcpu_run.c

```diff
@@ -598,7 +582,7 @@ static inline void km_vcpu_handle_pause(km_vcpu_t* vcpu, int hc_ret)
       km_cond_wait(&machine.pause_cv, &machine.pause_mtx);
       vcpu->state = HYPERCALL;
    }
-   km_mutex_unlock(&machine.pause_mtx);
+   pthread_cleanup_pop(1);   // km_mutex_unlock(&machine.pause_mtx);
```

Comment:
what is the reason for using cleanup_pop instead of using regular mutex_unlock?

Reply:
We do need to do the cleanup, and the unlock. I could do `km_mutex_unlock(&machine.pause_mtx); pthread_cleanup_pop(0);` - boils down to the same thing AFAIK

Reply:
Go it. Thanks.